src/integration/java/io/pinecone/integration/dataPlane/ListEndpointTest.java

```diff
@@ -0,0 +1,78 @@
+package io.pinecone.integration.dataPlane;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.pinecone.clients.AsyncIndex;
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.helpers.TestIndexResourcesManager;
+import io.pinecone.proto.ListResponse;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ListEndpointTest {
+    private static final TestIndexResourcesManager indexManager = TestIndexResourcesManager.getInstance();
+    private static String indexName ;
+    private static Index indexConnection;
+    private static AsyncIndex asyncIndexConnection;
+
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        indexName = indexManager.getServerlessIndexName();
+        indexConnection = indexManager.getServerlessIndexConnection();
+        asyncIndexConnection = indexManager.getServerlessAsyncIndexConnection();
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        String apiKey = System.getenv("PINECONE_API_KEY");
+        Pinecone pinecone = new Pinecone.Builder(apiKey).build();
+        pinecone.deleteIndex(indexName);
+        indexConnection.close();
+        asyncIndexConnection.close();
+    }
+
+    @Test
+    public void testListEndpoint() throws InterruptedException {
+        Thread.sleep(10000); // wait for the index to be ready for operations
```

Comment:
We should be calling `waitUntilIndexReady()` with an additional `Thread.sleep()` in the `TestIndexResourcesManager.createOrGetServerlessIndex()` so we really shouldn't need to wait again here. I think you can probably pull this out unless you're seeing issues.

Reply:
True, ty!